immutabledict/__init__.py

```diff
@@ -35,9 +35,6 @@ def __getitem__(self, key: _K) -> _V:
     def __contains__(self, key: object) -> bool:
         return key in self._dict
 
-    def copy(self, **add_or_replace: _V) -> immutabledict[_K, _V]:
-        return self.__class__(self, **add_or_replace)
-
```

Comment:
I think it'd be fairly type-safe if it constructed an `immutabledict` rather than `self.__class__`, e.g.

```python
class Strict(immutabledict[int, str]):
  pass

strict_d = Strict()
lenient_d: immutabledict[int, str | int] = strict_d  # so far so good

d = lenient_d.copy(foobar=42)
assert isinstance(d, Strict)
assert isinstance(d['foobar'], int)  # :-(
```

Reply:
I am fine keeping copy, pylance wanted me to remove the `_V` input type. Do you want me to add it back in?

Reply:
I'm not a maintainer, but I think if you want to make `copy` type-safe, you need to make it construct `immutabledict` rather than `self.__class__`.

Reply:
Thanks for the MR @spacether!

We could keep .copy() but without the extra parameters (they actually differs from dict.copy method, and I would prefer to stay closer to the dict implementation). It would fix the typing issue too.
Something like:
```
def copy(self, ) -> immutabledict[_K, _V]:
        return self.__class__(self,)
```

Can you update the MR if you don't mind? Else I can re-add the method later on my side.

Reply:
I'd say that would be a breaking change, though making `copy` argumentless may be a good thing to do. To reduce friction, I can propose the following:

```
def copy(self) -> immutabledict[_K, _V]: ...
def copy_update(self, **extra: _U) -> immutabledict[_K | str, _V | _U]: ...
```

So that migration path for users is at least as simple as "replace `.copy` with `.copy_update` if you need to update new dictionary, use `.copy` for only copying". Note how I added `_K | str` - if I do the following, current types would result in incorrect type:

```
foo: immutabledict[int, int]
bar = foo.copy(x=1)
reveal_type(bar)  # `bar` is `immutabledict[int, int]` - but it has `"x"` in keys!
```

Reply:
Thanks for the informations @sterliakov !
But yes I was thinking of making `copy` argumentless, it doesn't seem very necessary and so this PR seems to be a great opportunity to remove it.

Reply:
I added copy back in and it is argumentless